### ISSUE TITLE
normalize preprocess for the future use of pipeline

### ISSUE THREAD

**Issue by Qi-max** [hackingmaterials/automatminer#20]
For the future use of pipeline, the classes in `matbench.preprocess` can also take the form of `fit` + `transform`, as the classes in`sklearn.preprocessing`, such as `Imputer` and `LabelEncoder` etc.
